src/Analysis/Engine/Impl/Indexing/IndexerWalker.cs

```diff
@@ -0,0 +1,245 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.PythonTools.Parsing.Ast;
+
+namespace Microsoft.PythonTools.Analysis.Indexing {
+    internal class IndexerWalker : PythonWalker {
+        private readonly PythonAst _ast;
+
+        private readonly Stack<HierarchicalSymbol> _scope = new Stack<HierarchicalSymbol>();
+        private HierarchicalSymbol Parent => _scope.Count > 0 ? _scope.Peek() : null;
+
+        private bool _skipNext;
+        private bool SkipNext {
+            get {
+                var v = _skipNext;
+                _skipNext = false;
+                return v;
+            }
+            set { _skipNext = value; }
+        }
+
+        public List<HierarchicalSymbol> Symbols = new List<HierarchicalSymbol>();
```

Comment:
Public things should be readonly interfaces

Reply:
I switched it to a `IReadOnlyList`.